Implement range.count

nficca opened the pull request:
Implements `count` for #293.

windelbouwman commented on vm/src/obj/objrange.rs:
```diff
@@ -224,3 +233,20 @@ fn range_index(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
+    );
+
+    if let PyObjectPayload::Range { ref range } = zelf.borrow().payload {
+        match needle.borrow().payload {
```
Instead of using the PyObjectPayload here, use the vm.isinstance type check to determine int type, and objint::get_value to retrieve the BigInt value.

nficca replied:
This works for the argument, but won't I still eventually need to use `PyObjectPayload` to pull out the actual `RangeType`? Apologies if I misunderstood you, I'm not entirely familiar with this project.

windelbouwman replied:
No worries, thank you for your help! The idea is to use PyObjectPayload as little as possible. The objrange.rs file should contain a method get_value which will retrieve the payload for you. This saves a lot of duplicate code for borrow and matching on the payload type. Take a look at the objint::getvalue method for example. It returns you the BigInt contents of the integer. The same should be there for the range type.

Hope this helps!

windelbouwman replied:
Btw, the problem with checking strictly on the payload is that this will eventually fail with subclasses. The isinstance check is more thorough and works for subclasses as well!

nficca commented:
@windelbouwman I didn't find the `objrange::get_value` method you mentioned so I've added it myself. I make use of it and `objtype::getinstance` as you suggested in `range_count`; I *think* this is what you meant?

Also, I modified a bunch of the other methods in the file to make use of `objrange::get_value` as well. Maybe this is beyond the scope of this PR... If so I can move those changes to a separate PR.

windelbouwman commented:
This is exactly what I meant. Great job!
